pkg/metrics/scraper_test.go

```diff
@@ -0,0 +1,133 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/client-go/rest"
+)
+
+func TestScrapeAndParse(t *testing.T) {
+	metricsServer := newMockPrometheusServer()
+	server := httptest.NewServer(&metricsServer)
+	defer server.Close()
+	tests := []struct {
+		name                  string
+		localClusterNodes     int
+		skipLocalLabel        bool
+		numOtherClusters      int
+		nodesPerOtherClusters int
+		skipOtherLabel        bool
+		authed                bool
+		expectedTotal         int
+		expectedError         bool
+	}{
+		{
+			name:                  "no downstream, local only",
+			localClusterNodes:     2,
+			skipLocalLabel:        false,
+			numOtherClusters:      0,
+			nodesPerOtherClusters: 0,
+			skipOtherLabel:        false,
+			authed:                true,
+			expectedTotal:         0,
+			expectedError:         false,
+		},
+		{
+			name:                  "downstream and local",
+			localClusterNodes:     2,
+			skipLocalLabel:        false,
+			numOtherClusters:      1,
+			nodesPerOtherClusters: 2,
+			skipOtherLabel:        false,
+			authed:                true,
+			expectedTotal:         2,
+			expectedError:         false,
+		},
+		{
+			name:                  "multiple downstream and local",
+			localClusterNodes:     2,
+			skipLocalLabel:        false,
+			numOtherClusters:      3,
+			nodesPerOtherClusters: 4,
+			skipOtherLabel:        false,
+			authed:                true,
+			expectedTotal:         12,
+			expectedError:         false,
+		},
+		{
+			name:                  "local not labeled, local nodes excluded",
+			localClusterNodes:     2,
+			skipLocalLabel:        true,
+			numOtherClusters:      0,
+			nodesPerOtherClusters: 0,
+			skipOtherLabel:        false,
+			authed:                true,
+			expectedTotal:         0,
+			expectedError:         false,
+		},
+		{
+			name:                  "local not labeled, downstream not labeled, all nodes excluded",
+			localClusterNodes:     2,
+			skipLocalLabel:        true,
+			numOtherClusters:      2,
+			nodesPerOtherClusters: 2,
+			skipOtherLabel:        true,
+			authed:                true,
+			expectedTotal:         0,
+			expectedError:         false,
+		},
+		{
+			name:                  "local labeled, downstream not labeled, all nodes excluded",
+			localClusterNodes:     2,
+			skipLocalLabel:        false,
+			numOtherClusters:      2,
+			nodesPerOtherClusters: 2,
+			skipOtherLabel:        true,
+			authed:                true,
+			expectedTotal:         0,
+			expectedError:         false,
+		},
+		{
+			name:                  "error, no auth",
+			localClusterNodes:     2,
+			skipLocalLabel:        false,
+			numOtherClusters:      0,
+			nodesPerOtherClusters: 0,
+			skipOtherLabel:        false,
+			authed:                false,
+			expectedTotal:         0,
+			expectedError:         true,
+		},
+	}
+	for _, test := range tests {
+		test := test
```

Comment:
I don't think you'd need this, since all the code in the loop is synchronous.

Reply:
Removed this and it seemed to work ok- so I think we are good to remove it.